### PR TITLE
Add failing test for raycasting accuracy on `ConvexPolygon`

### PR DESCRIPTION
**Pull request by Ameobea**
* Creates a square `ConvexPolygon` and attempts to find collisions between it and a point behind it, adjusting the origin of the ray each time.

### CODE REVIEW

File: build/ncollide2d/tests/geometry/ray_cast.rs [hunk 1]
```diff
@@ -88,3 +88,54 @@ fn perpendicular_raycast_starting_bellow_segment() {
...
+        .collect();
+
+    let square = ConvexPolygon::try_new(vertices).unwrap();
+    let raycaster = square.as_ray_cast().unwrap();
+
+    let test_raycast = |ray_origin: Point2<f64>, ray_look_at: Point2<f64>| -> Option<f64> {
+        let mut ray_angle: Vector2<f64> = ray_look_at - ray_origin;
+        let ray_angle_magnitude =
+            ((ray_angle.x * ray_angle.x) + (ray_angle.y + ray_angle.y)).sqrt();
+        ray_angle /= ray_angle_magnitude;
```
Comment (sebcrozet) [inline build/ncollide2d/tests/geometry/ray_cast.rs]: Line 115 to 117: there is already a feature on nalgebra to do this the normalization: `ray_angle.normalize()`. So you could just use: `Ray::new(ray_origin, ray_angle.normalize())` at line 121.

File: build/ncollide2d/tests/geometry/ray_cast.rs [hunk 2]
```diff
@@ -88,3 +88,54 @@ fn perpendicular_raycast_starting_bellow_segment() {
...
+/// 2  *--+  |
+/// |  |  |  +
+/// 1  +--+  *<----+Ray origin
+/// |
+/// 0--1--2--3
+///
+/// Tests the accuracy of raycaster collision detection against a `ConvexPolygon`.
+#[test]
+fn convexpoly_raycast_fuzz() {
+    let vertices: Vec<Point2<f64>> = vec![[1, 1], [1, 2], [2, 2], [2, 1]]
```
Comment (sebcrozet) [inline build/ncollide2d/tests/geometry/ray_cast.rs]: This should be `[1, 1], [2, 1], [2, 2], [1, 2]`. This is not mentioned on the doc (I'll fix that) but the vertices must be provided in counterclockwise order.

**Comment by Ameobea** [comment]
@sebcrozet Thanks for the tip about `.normalize()`; I didn't know about that.  I've also reversed the point order for the edges.

**Comment by sebcrozet** [comment]
Hi @Ameobea! Thank you for the reminder. This will be merged only after the issue is fixed.

**Comment by Ralith** [comment]
Maybe it should be merged but marked with `#[ignore]` until fixed? That way it wouldn't need to be constantly rebased against churn.

**Comment by sebcrozet** [comment]
I didn't know about `#[ignore]` so that would have been a good idea, yes.
This issue will be fixed by #300 which will also include the test provided in this PR.

So I'll close this now. Thank you @Ameobea for providing this test case!

**Comment by Ameobea** [comment]
Awesome - that's great to hear! :D

**Pull request closed without merging**